src/functions/template-tags/html.php

```diff
@@ -132,3 +132,83 @@ function tribe_disabled( $disabled, $echo = true ) {
 		}
 	}
 }
+
+function tribe_format_field_dependency( $deps ) {
+    // Sanity check.
+    if ( empty( $deps ) ) {
+        return '';
+	}
+
+	// Let's be case-insensitive!
+	if ( isset( $deps['ID'] ) ) {
+		$deps['id'] = $deps['ID'];
+		unset( $deps['ID'] );
+	}
+
+    // No ID to hook to? Bail.
+    if ( empty( $deps['id'] ) ) {
+        return;
+    }
+
+    $dependency = '';
+
+    $accepted = [
+        'id',
+        'is',
+        'is-not',
+        'is-empty',
+        'is-not-empty',
+        'is-numeric',
+        'is-not-numeric',
+        'is-checked',
+        'is-not-checked',
+    ];
+
+    foreach ( $deps as $attr => $value ) {
+        // Attributes are always lower case.
```

Comment:
Since you're assuming lower-case names here, I would just make all keys lowercase at the start of the function:

```php
$deps = array_combine( array_map( 'strtolower', array_keys( $deps ) ), $deps );
```

And not bother with upper or lowercase keys again in the method.